sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "context.hpp"
+#include "http.hpp"
+
+namespace Azure
+{
+namespace Core
+{
+namespace Http
+{
+
+class HttpPolicy
+{
+public:
+  // If we get a response that goes up the stack
+  // Any errors in the pipeline throws an exception
+  // At the top of the pipeline we might want to turn certain responses into exceptions
+  virtual Response Process(Context& context, Request& request) const = 0;
+  virtual ~HttpPolicy() {}
+};
+
+class HttpTransport : public HttpPolicy
+{
+  Response Process(Context& context, Request& request) const override
+  {
+    context.CancelWhen();
+    request.getHeaders();
+
+    return Response(200, "OK\n");
+  }
+};
+
+struct RetryPolicyOptions
```

Comment:
Seems like this could just be an aggregate?

Reply:
I'd just call this RetryOptions. Policy should not be in front of the customers face.